### PR TITLE
Update checkbox of organisation and observation type by updating Redux store

### PR DESCRIPTION
**Pull request by hoanphungt**
Fix the checkbox updating mechanism in Lizard Catalogue by properly update Redux store with actions dispatched

### CODE REVIEW

File: src/components/FilterBar.tsx [hunk 1]
```diff
@@ -153,13 +150,17 @@ class FilterBar extends React.Component<MyProps, MyState> {
...
+                            {checkedOrganisation.name}
+                        </div>
+                        :
                         <div className="filter__checked-item" />
                     }
                     <ul className="filter-list">
                         {filteredOrganisations.slice(0, this.state.orgItems).map((organisation: Organisation) => (
                             <li className="filter-item" key={organisation.uuid}>
-                                <input type="checkbox" className="filter-checkbox" onClick={() => onOrganisationCheckbox(organisation)} onChange={() => this.onOrganisationChange(organisation, organisations)} checked={organisation.checked} />
+                                <input type="checkbox" className="filter-checkbox" onChange={() => onOrganisationCheckbox(organisation)} checked={organisation.checked} />
```
Comment (GeOdin) [inline src/components/FilterBar.tsx]: Nice way to make an interactive checkbox! :)

File: src/action.ts [hunk 2]
```diff
@@ -179,6 +181,27 @@ export const fetchOrganisations = (dispatch: Dispatch<OrganisationsFetched>): vo
...
+const organisationCheckboxUpdated = (organisation: Organisation): UpdateOrganisationCheckbox => ({
+    type: UPDATE_ORGANISATION_CHECKBOX,
+    payload: organisation
+});
+
+export const updateOrganisationCheckbox = (organisation: Organisation, dispatch: Dispatch<UpdateOrganisationCheckbox>) => {
+    dispatch(organisationCheckboxUpdated(organisation));
+};
+
+const observationCheckboxUpdated = (observationType: ObservationType): UpdateObservationCheckbox => ({
```
Comment (GeOdin) [inline src/action.ts]: Would it be an idea to name this variable ```observationTypeCheckboxUpdated``` to keep it consistent with observationType?
I was confused whether there was something else than observationType that was observation.
There are more examples where observation was used, I would suggest to user observationType for more consistency.

```
const observationTypeCheckboxUpdated = (observationType: ObservationType): UpdateObservationTypeCheckbox => ({
    type: UPDATE_OBSERVATIONTYPE_CHECKBOX,
    payload: observationType
});

export const updateObservationTypeCheckbox = (observationType: ObservationType, dispatch: Dispatch<UpdateObservationTypeCheckbox>) => {
    dispatch(observationTypeCheckboxUpdated(observationType));
};
```

File: src/components/FilterBar.tsx [hunk 3]
```diff
@@ -13,7 +13,9 @@ interface MyProps {
     fetchWMSLayers: (page: number, searchTerm: string, organisationName: string, ordering: string) => void,
     onObservationTypeCheckbox: (obsType: ObservationType) => void,
     onOrganisationCheckbox: (organisation: Organisation) => void,
-    onViewChange: () => void
+    updateObservationCheckbox: (obsType: ObservationType) => void,
```
Comment (GeOdin) [inline src/components/FilterBar.tsx]: I notice that sometimes you use 
```(obsType: ObservationType) ...```
and other times you use
```(observationType: ObservationType)``` (in action.tx)
Would it be an idea to use the same name, to make it easier to understand that these are the same and to improve consistency?

File: src/action.ts (hunk 2 above)
Comment (hoanphungt) [reply src/action.ts]: Thanks Madeleine, I changed it to use ObservationTypeCheckbox everywhere

**Pull request merged**